src/main/java/com/cloudbees/jenkins/plugins/bitbucket/BitbucketSCMSource.java

```diff
@@ -603,6 +604,50 @@ protected void retrieve(@CheckForNull SCMSourceCriteria criteria, @NonNull SCMHe
         }
     }
 
+    /**
+     * Generates branch names based on the pull request naming strategy. Behavior is governed by the {@link PullRequestNamingTrait}.
+     *
+     * @param pullRequest          {@link BitbucketPullRequest}.
+     * @param originalBranchName   the original branch name of the pull request.
+     * @param activeStrategyCount  the number of active pull request discovery strategies, see {@link ChangeRequestCheckoutStrategy}.
+     * @param strategy             the current {@link ChangeRequestCheckoutStrategy}.
+     * @param namingStrategy       the {@link PullRequestNamingStrategy}.
+     * @param namingExcludePattern the naming exclude {@link Pattern}.
+     * @return the new branch name.
+     * @since 3.6.0
+     */
+    public static String applyPRsNamingStrategy(final BitbucketPullRequest pullRequest, String originalBranchName,
+                                                int activeStrategyCount, ChangeRequestCheckoutStrategy strategy,
+                                                PullRequestNamingStrategy namingStrategy, Pattern namingExcludePattern) {
+        StringBuilder branchName = new StringBuilder();
```

Comment:
same here about `branchName` variable name

Reply:
I changed it to `pullDisplayName` which seems to be more in line with the other var names.